impeller/display_list/dl_dispatcher.cc

```diff
@@ -36,6 +36,53 @@
 
 namespace impeller {
 
+#define USE_DEPTH_WATCHER true
+
+#if USE_DEPTH_WATCHER
+struct DepthWatcher {
+  DepthWatcher(const std::string& file,
+               int line,
+               const impeller::Canvas& canvas,
+               int allowed)
+      : file_(file),
+        line_(line),
+        canvas_(canvas),
+        allowed_(allowed),
+        old_depth_(canvas.GetOpDepth()),
+        old_max_(canvas.GetMaxOpDepth()) {}
+
+  ~DepthWatcher() { check(file_, line_); }
+
+  void check(const std::string& file, int line) {
+    FML_CHECK(canvas_.GetOpDepth() <= (old_depth_ + allowed_) &&
+              canvas_.GetOpDepth() <= old_max_)
+        << std::endl
+        << "from " << file << ":" << line << std::endl
+        << "old/allowed/current/max = " << old_depth_ << "/" << allowed_ << "/"
+        << canvas_.GetOpDepth() << "/" << old_max_;
+  }
+
+ private:
+  const std::string file_;
+  const int line_;
+
+  const impeller::Canvas& canvas_;
+  const uint64_t allowed_;
+  const uint64_t old_depth_;
+  const uint64_t old_max_;
+};
+
+#define AUTO_DEPTH_WATCHER(d) \
```

Comment:
Can you document, here or elsewhere how to use AUTO_DEPTH_WATCHER if I were to add a new canvas command. it just takes the required depth increment, right?

Reply:
Done